packages/types/src/lib/interfaces.ts

```diff
@@ -1891,3 +1901,34 @@ export interface SignatureData {
 export type ClaimsList = {
   [key: string]: SignatureData;
 }[];
+
+export interface MintWithAuthParams {
+  /**
+   * auth method to use for minting
+   */
+  authMethod: AuthMethod;
+
+  /**
+   * Permission scopes:
+   * https://developer.litprotocol.com/v3/sdk/wallets/auth-methods/#auth-method-scopes
+   */
+  scopes: string[] | number[];
+
+  /**
+   * only applies to webauthn auth method
+   */
+  pubkey?: string;
+
+  /**
+   * The Auth ID of the given auth method. If it's custom auth, then it could be
+   * anything.
+   */
+  authId?: Uint8Array;
+}
+
+export interface mintWithCustomAuthParams extends MintWithAuthParams {
+  /**
+   * For a custom authentication method, the custom auth ID should uniquely identify the user for that project. For example, for Google, we use appId:userId, so you should follow a similar format for Telegram, Twitter, or any other custom auth method.
+   */
+  customAuthId: string | Uint8Array;
```

Comment:
Should we also support other authMethods when using custom auth? I don't think we can right now if we only support a single `id`